pkg/pool-weighted/contracts/WeightedPool2Tokens.sol

```diff
@@ -800,7 +341,8 @@ contract WeightedPool2Tokens is
         uint256 balanceToken0,
         uint256 balanceToken1
     ) internal {
-        bytes32 miscData = _miscData;
+        bytes32 miscData = _getMiscData();
+
         if (miscData.oracleEnabled() && block.number > lastChangeBlock) {
```

Comment:
```suggestion
        if (block.number <= lastChangeBlock) {
            return;
        }
        
        bytes32 miscData = _getMiscData();

        if (miscData.oracleEnabled()) {
```

Seems like we can save an SLOAD for pools with the oracle disabled here.

Reply:
Note that the SLOAD would be warm because we also keep the swap fee percentage there, but this is still an improvement. Thanks!